strawberryfields/api_client.py

```diff
@@ -216,6 +218,35 @@ def join_path(self, path):
         """
         return join_path(self.BASE_URL, path)
 
+    def request(self, method, **params):
+        """
+        Calls ``method`` with ``params`` after applying headers. Records the request type and
+        parameters to ``self.errors`` if the request is not successful, and the response to
+        ``self.responses`` if a response is returned from the server.
+
+        Args:
+            method: one of ``requests.get`` or ``requests.post``
+            **params: the parameters to pass on to the method (e.g. ``url``, ``data``, etc.)
+
+        Returns:
+            requests.Response: a response object, or None if no response could be fetched
+        """
+        assert method in (requests.get, requests.post)
```

Comment:
Under what circumstances could this assertion fail?

Also, is this assertion only performed in debug mode?

Reply:
(similarly to how running Python with the `-O` flag ignores assertions)

Reply:
This is mostly superfluous, although I suppose that if a developer was using this method and passing something other than `requests.get` or `requests.post`, a more ambiguous error might arise, or this could fail silently and give an unexpected result (i.e. by calling whatever method is passed with `params`, which could return a very unexpected result and still succeed.) It's mostly for future developers. So to answer your question: it would not play a part or be important or significant if we are ignoring assertions.

Though more suitably perhaps this should raise a `TypeError`, so I will change.